### PR TITLE
[DDL-yugabyte] Adding workarounds for Yugabyte to make the DbMigrationTest passing

### PR DESCRIPTION
**Pull request by rPraml**
This adds workarounds for Yugabyte-DDL generation.
in Yugabyte, after a DML statement, (insert, update, delete), the transaction has to be commited.

I'm considering if the commit statements should be added automatically after insert/uptate/delete when generating the yugabyte-DDL file:

- adding `boolean needsDmlCommit()` to PlatformDdl (which returns true for YB)
- detecting, if we have an IUD statement and add commt after the statements (somewhere here https://github.com/ebean-orm/ebean/blob/master/ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/BaseTableDdl.java#L170)

On the other side, this could/should be also fixed in yugabyte: https://github.com/yugabyte/yugabyte-db/issues/3109

@rbygrave what do you think?

### CODE REVIEW

File: ebean-test/src/test/java/misc/migration/v1_1/EBasic.java [hunk 1]
```diff
@@ -41,6 +41,10 @@
   Integer id;
 
   @NotNull
+  @DbMigration(platforms = Platform.YUGABYTE, preAlter = {
+      "update ${table} set ${column} = ${default} where ${column} is null",
+      "commit transaction /* 1 */" })
```
Comment (rPraml) [inline ebean-test/src/test/java/misc/migration/v1_1/EBasic.java:46]: yugabyte needs commits here
@rbygrave what do you think, if the ddl generator will handle the case. e.g. here: https://github.com/ebean-orm/ebean/blob/master/ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/BaseTableDdl.java#L130

File: ebean-test/src/test/resources/migrationtest/dbmigration/yugabyte/1.1.sql [hunk 2]
```diff
@@ -55,13 +56,21 @@ alter table migtest_e_basic alter column status2 drop not null;
 alter table migtest_e_basic add constraint uq_migtest_e_basic_description unique  (description);
 
 insert into migtest_e_user (id) select distinct user_id from migtest_e_basic;
+commit transaction /* 3 */;
```
Comment (rPraml) [inline ebean-test/src/test/resources/migrationtest/dbmigration/yugabyte/1.1.sql:59]: @rbygrave what is the reason, that the duplicate lines are skipped in DdlRunner?

Comment (rbygrave) [reply ebean-test/src/test/resources/migrationtest/dbmigration/yugabyte/1.1.sql:59]: Not sure.  What is the code that does that?

**Comment by rbygrave** [comment]
I don't think we should rush to put in workarounds for what are bugs in Yugabyte (or Cockroach).

**Comment by rPraml** [comment]
I agree, as this PR contains no real fixes, it should not be merged hastily.

Nearly all workarounds are done with modified DbMigration annotations in src/test and the one in src/main could be also covered by a proper DbMigration annotation.

Should I keep that PR open (for documentation)?

The other question about duplicates targets this code https://github.com/ebean-orm/ebean-ddl-runner/blob/7137006f9e3ab97a323db3f43139bfd6349f002a/src/main/java/io/ebean/ddlrunner/DdlRunner.java#L67
Why is it necessary to have unique statements? What was the intention and can/should this behavior changed in DdlRunner? 
There is also a workaround-counter variable in HanaHistoryDdl and we also fell into that trap with hand written scripts.

Cheers
Roland

File: ebean-test/src/test/resources/migrationtest/dbmigration/yugabyte/1.1.sql (hunk 2 above)
Comment (rbygrave) [reply ebean-test/src/test/resources/migrationtest/dbmigration/yugabyte/1.1.sql:59]: Hmmm. No git history there, it has been moved from ddl-generator I believe.  Now I'd say that isn't necessary and we can remove that `noDuplicates` list altogether.

**Comment by rbygrave** [comment]
Lets keep this around for documentation for a while and see how we feel in a few weeks.

> DdlRunner.java#L67

I've removed that code and released `ebean-ddl-runner 1.3`

**Comment by rbygrave** [comment]
Closing this for now, added comments into  https://github.com/ebean-orm/ebean/issues/2510

**Pull request closed without merging**